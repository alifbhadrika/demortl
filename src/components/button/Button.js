import React, { useState } from "react";

/* Button component, will render pop up 
text to the page when it gets clicked */

function Button({ popUpText }) {
  const [showText, setShowText] = useState(false);
  const handleOnClick = () => {
    setShowText(!showText);
  };
  return (
    <div className="button">
      <button
        data-testid="BUTTON_ID"
        style={{ padding: 5, borderRadius: 16 }}
        onClick={handleOnClick}
      >
        Click Me!
      </button>
      {showText && <div style={{ marginTop: 20 }}>{popUpText}</div>}
    </div>
  );
}

export default Button;
