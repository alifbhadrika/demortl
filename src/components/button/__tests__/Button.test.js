import React from "react";
import Button from "../Button";
import { render, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("Tests for Button component", () => {
  it("snapshot button component", () => {
    const mockPopUpText = "This is just for the sake of the test";
    const { asFragment } = render(<Button popUpText={mockPopUpText} />);
    expect(asFragment(<Button popUpText={mockPopUpText} />)).toMatchSnapshot();
  });

  it("should render button component", () => {
    const mockPopUpText = "This is just for the sake of the test";
    const { getByText } = render(<Button popUpText={mockPopUpText} />);
    expect(getByText("Click Me!")).toBeInTheDocument();
  });

  it("we can also find by the role button", () => {
    const mockPopUpText = "This is just for the sake of the test";
    const { getByRole } = render(<Button popUpText={mockPopUpText} />);
    expect(getByRole("button")).toBeInTheDocument();
  });

  it("we can also find by the test id", () => {
    const mockPopUpText = "This is just for the sake of the test";
    const { getByTestId } = render(<Button popUpText={mockPopUpText} />);
    expect(getByTestId("BUTTON_ID")).toBeInTheDocument();
  });

  it("doesn't shows pop up text when it never gets clicked", () => {
    const mockPopUpText = "This is just for the sake of the test";
    const { queryByText } = render(<Button popUpText={mockPopUpText} />);
    expect(queryByText(mockPopUpText)).toBeNull();
  });

  it("shows pop up text when it gets clicked", () => {
    const mockPopUpText = "This is just for the sake of the test";
    const { getByText, getByRole } = render(
      <Button popUpText={mockPopUpText} />
    );
    const myButton = getByRole("button");

    userEvent.click(myButton);
    expect(getByText(mockPopUpText)).toBeInTheDocument();
  });

  it("doesn't shows pop up text when it gets clicked twice", () => {
    const mockPopUpText = "This is just for the sake of the test";
    const { queryByText, getByRole } = render(
      <Button popUpText={mockPopUpText} />
    );
    const myButton = getByRole("button");

    fireEvent.click(myButton);
    fireEvent.click(myButton);
    expect(queryByText(mockPopUpText)).not.toBeInTheDocument();
  });
});
