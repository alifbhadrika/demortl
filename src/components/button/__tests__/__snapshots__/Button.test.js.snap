// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Button component snapshot button component 1`] = `
<DocumentFragment>
  <div
    class="button"
  >
    <button
      data-testid="BUTTON_ID"
      style="padding: 5px; border-radius: 16px;"
    >
      Click Me!
    </button>
  </div>
</DocumentFragment>
`;
